packages/expo-dev-menu/ios/DevMenuAppInstance.swift

```diff
@@ -43,7 +43,11 @@ class DevMenuAppInstance: NSObject, RCTBridgeDelegate {
   func sourceURL(for bridge: RCTBridge!) -> URL! {
     #if DEBUG
     if let packagerHost = jsPackagerHost() {
-      return RCTBundleURLProvider.jsBundleURL(forBundleRoot: "index", packagerHost: packagerHost, enableDev: true, enableMinification: false)
+      return RCTBundleURLProvider.jsBundleURL(
+        forBundleRoot: ".expo/.virtual-metro-entry",
+        packagerHost: packagerHost,
+        enableDev: true,
+        enableMinification: false)
```

Comment:
That's fine, our Metro config should be an absolute requirement for working with Expo tools. It's needed for things like [aliases and absolute imports](https://docs.expo.dev/guides/typescript/#path-aliases), asset hashing, web support, arbitrary entry file, bundle splitting, file-based routing, etc.

Reply:
@EvanBacon @lukmccall I just got hit with this. I use react-native-esbuild which is broken by this change. Since this change _any_ development build strictly requires using metro with expo's config because `.expo/.virtual-metro-entry` gets hardcoded into the app at build time. 

Perhaps a workaround could be added to fall back to the default `index.js` if `.expo/.virtual-metro-entry` fails to resolve?

Reply:
although it looks like adding a file `.expo/.virtual-metro-entry.js` that requires my index.js can work as a workaround